tests/userController.ts

```diff
@@ -90,4 +97,55 @@ describe('Unit testing of user controllers', function() {
 		expect(resSpy.calledOnce).to.equal(true);
 		expect(resSpy.args[0][0]).to.eql({ downvotes: {}, upvotes: {} });
 	});
+
+	it('should get public user data', async function () {
+		const resSpy = sinon.spy();
+		const res = {
+			send: resSpy
+		};
+		await userController.getUserData(req, res);
+		expect(resSpy.calledOnce).to.equal(true);
+		expect(resSpy.args[0][0]).to.eql({
+			username: 'testUsername',
+			upvotes: 0,
+			downvotes: 0
+		});
+	});
+	describe('Unit testing icons', function () {
+		it('should change user icon', async function () {
+			const goodReq = {
+				file: {
+					path: 'testPath'
+				},
+				cookies: {
+					username: 'testUsername'
+				}
+			};
+			const badReq = {
+				cookies: {
+					username: 'testUsername'
+				}
+			};
+			const resSpy = sinon.spy();
+			const res = {
+				sendStatus: resSpy
+			};
+			await userController.changeUserIcon(goodReq, res);
+			expect(resSpy.calledOnce).to.equal(true);
+			expect(resSpy.args[0][0]).to.equal(200);
+			await userController.changeUserIcon(badReq, res);
+			expect(resSpy.args[1][0]).to.equal(400);
+		});
+		it('should get user icon', async function () {
+			const resSpy = sinon.spy();
+			const res = {
+				sendFile: resSpy
+			};
+			await userController.getUserIcon({ params: 'testUsername' }, res);
+			expect(resSpy.calledOnce).to.equal(true);
+			expect(resSpy.args[0][0]).to.equal('testPath');
+			await userController.getUserIcon({ params: 'testUsername' }, res);
+			expect(resSpy.args[1][0]).to.contain('default.png');
```

Comment:
I'm trying to get an idea of this test.
What is the difference between calls? 
Hint: if another can't get an idea, most likely that it depends on different mock responses and it would be better to mock them in this same test.
Or just add the comments :)

Reply:
Yes, basically getUserIcon will alternate between returning null and an actual response for testing. I will add this comment here